### PR TITLE
[DEVELOPER_GUIDE]  resolving links

### PR DESCRIPTION
**Pull request by Aigerim-ai**
following the comment  by @joshuarrrr "I think the DEVELOPER_GUIDE is a more natural location for both of these links, particularly for plugin generation, which is a dev-only task.

My suggestion is to add a new section to the DEVELOPER_GUIDE for "Plugin development", and start with a list of resources there."

This PR adds new section for DEVELOPER_GRUIDE

fixes : https://github.com/opensearch-project/OpenSearch-Dashboards/issues/778

### CODE REVIEW

File: DEVELOPER_GUIDE.md [hunk 1]
```diff
@@ -227,6 +228,10 @@ $ yarn opensearch snapshot --P https://repo1.maven.org/maven2/org/opensearch/plu
 
 Note - if you add the [`security` plugin](https://github.com/opensearch-project/security), you'll also need to [configure OpenSearch Dashboards for security](#configure-opensearch-dashboards-for-security).
 
+### Plugin development
+
+[osd-plugin-generator]: (https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/packages/osd-plugin-generator)
```
Comment (ashwin-pc) [inline DEVELOPER_GUIDE.md:233]: I dont think this works as you expect. This is not the correct format for markdown links. Markdown links are formatted as `[Link name](URL)`, no colon in between. Also can you add some content in here about the link you are pointing the users to?

**Review by ashwin-pc — COMMENTED** [review]
Thanks for the change @Aigerim-ai, a few changes:
1. Can you add a changelog entry for this change?
2. Can you add some content in the Plugin Development section so that the users know why they need to go to the link you added
3. Can you review the content in `packages/osd-plugin-generator/README.md` to make sure that it is still accurate?

File: DEVELOPER_GUIDE.md (hunk 1 above)
Comment (Aigerim-ai) [reply DEVELOPER_GUIDE.md]: Done, could you please review my changes?

File: DEVELOPER_GUIDE.md [hunk 2]
```diff
@@ -227,6 +228,13 @@ $ yarn opensearch snapshot --P https://repo1.maven.org/maven2/org/opensearch/plu
 
 Note - if you add the [`security` plugin](https://github.com/opensearch-project/security), you'll also need to [configure OpenSearch Dashboards for security](#configure-opensearch-dashboards-for-security).
 
+### Plugin development
+The osd-plugin-generator tool simplifies the process of creating an OpenSearch Dashboards plugin by setting up the initial project structure and providing build scripts. It enables developers to focus on the core functionality and customization of their plugins, thereby enhancing the data visualization and management capabilities of OpenSearch Dashboards.
+
+The osd-plugin-generator is a tool available in the OpenSearch Dashboards repository. It allows you to generate an OpenSearch Dashboards plugin. OpenSearch Dashboards is an open-source project that provides a web interface for data visualization and management in the OpenSearch ecosystem.
+[osd-plugin-generator](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/packages/osd-plugin-generator)
```
Comment (ashwin-pc) [inline DEVELOPER_GUIDE.md:235]: Lets keep it simple here for now

```suggestion
The osd-plugin-generator tool makes it easier to create a plugin for OpenSearch Dashboards. It sets up the basic structure of the project and provides scripts to build it. Refer to [osd-plugin-generator](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/packages/osd-plugin-generator) for more details.
```

File: DEVELOPER_GUIDE.md [hunk 3]
```diff
@@ -231,6 +232,10 @@ $ yarn opensearch snapshot --P https://repo1.maven.org/maven2/org/opensearch/plu
 
 Note - if you add the [`security` plugin](https://github.com/opensearch-project/security), you'll also need to [configure OpenSearch Dashboards for security](#configure-opensearch-dashboards-for-security).
 
+### Plugin development
+The osd-plugin-generator tool makes it easier to create a plugin for OpenSearch Dashboards. It sets up the basic structure of the project and provides scripts to build it. Refer to [osd-plugin-generator](https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/packages/osd-plugin-generator) for more details.
+
```
Comment (joshuarrrr) [inline DEVELOPER_GUIDE.md:237]:
```suggestion
```

**Review by joshuarrrr — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Pull request merged**